File: rest/src/main/java/de/philgenstock/choremaster/user/UserEntity.java
package de.philgenstock.choremaster.user;

import de.philgenstock.choremaster.household.HouseHoldEntity;
import de.philgenstock.choremaster.persistence.BaseEntity;
import jakarta.persistence.*;
import java.util.ArrayList;
import java.util.List;
import lombok.Data;

@Data
@Table(name = "users")
@Entity
public class UserEntity extends BaseEntity {

  private String googleId;
  private String name;
  private String email;

  @ManyToMany(
      cascade = {CascadeType.ALL},
      fetch = FetchType.LAZY)
  @JoinTable(
      name = "users_to_household",
      joinColumns = {@JoinColumn(name = "user_id")},
      inverseJoinColumns = {@JoinColumn(name = "household_id")})
  private List<HouseHoldEntity> houseHolds = new ArrayList<>();

  public void addHouseHold(HouseHoldEntity houseHold) {
    houseHolds.add(houseHold);
  }
}

File: choremaster-rest/rest/src/main/kotlin/de/philgenstock/choremaster/user/UserEntity.kt
package de.philgenstock.choremaster.user

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.Instant

@Entity
@Table(name = "users")
data class UserEntity(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    @Column(name = "last_login")
    val lastLogin: Instant,
    @Column(name = "created_date")
    val createdDate: Instant,
    @Column(name = "first_name")
    val firstName: String,
    @Column(name = "last_name")
    val lastName: String,
    val email: String,
) {
    companion object
}
